Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIStorageAreaCocoa.mm

```diff
@@ -152,17 +152,28 @@
         return;
     }
 
-    if (!isValidJSONObject(items)) {
-        *outExceptionString = toErrorString(nil, @"items", @"it is not JSON-serializable");
-        return;
-    }
+    __block NSString *keyWithError;
+
+    auto *serializedData = mapObjects(items, ^NSString *(NSString *key, id object) {
+        if (keyWithError)
+            return nil;
+
+        auto *result = encodeJSONString(object, JSONOptions::FragmentsAllowed);
+        if (!result) {
+            keyWithError = key;
+            return nil;
+        }
 
-    NSDictionary<NSString *, NSString *> *serializedData = mapObjects(items, ^NSString *(NSString *key, id object) {
-        return encodeJSONString(object, { JSONOptions::FragmentsAllowed });
+        return result;
     });
 
-    if (anyItemsExceedQuota(serializedData, webExtensionStorageAreaSyncQuotaBytesPerItem)) {
-        *outExceptionString = toErrorString(nil, @"items", @"exceeded maximum size for a single item");
+    if (keyWithError) {
+        *outExceptionString = toErrorString(nil, [NSString stringWithFormat:@"items[`%@`]", keyWithError], @"it is not JSON-serializable");
```

Comment:
If I'm reading `toErrorString()` correctly, I think the resulting error string would read better without the `it`:

    *outExceptionString = toErrorString(nil, [NSString stringWithFormat:@"items[`%@`]", keyWithError], @"is not JSON-serializable");

Reply:
The full error turns into:

```
Invalid call to storageArea.set(). The 'items[`key`]' value is invalid, because it is not JSON-serializable.
```

So "it" is needed.